javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -398,7 +398,7 @@ class BigDecimal() extends Number with Comparable[BigDecimal] {
     }
 
     val (unscaled, bufLength) = {
-      val u = in.subSequence(begin, index).toString
+      val u = ArrayCharSequence(in).subSequence(begin, index).toString
```

Comment:
`ArrayCharSequence` will be deprecated in the next version. Let's avoid it altogether:
```suggestion
      val u = new String(in, begin, index - begin)
```